modules/transport-grpc/transport-grpc-spi/build.gradle

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+apply plugin: 'opensearch.build'
+apply plugin: 'opensearch.publish'
+
+base {
+  group = 'org.opensearch.plugin'
+  archivesName = 'transport-grpc-protobuf-spi'
+}
+
+dependencies {
+  api project(":server")
+  api "org.opensearch:protobufs:${versions.opensearchprotobufs}"
+  api "io.grpc:grpc-api:${versions.grpc}"
```

Comment:
This looks broadly right to me. Both of these are really lightweight dependencies. That fact that guava is a runtime dependency of grpc-api is fine because the runtime in this context will be the transport-grpc module which has grpc-core (and guava and everything else needed to actually run a grpc server). The things that will take a dependency on this SPI just need the minimal grpc API classes as well as the OpenSearch protobuf definitions.